test/test_base_metric.py

```diff
@@ -5,86 +5,108 @@
 from sklearn import clone
 from sklearn.utils.testing import set_random_state
 from test.test_utils import ids_metric_learners, metric_learners
+import string
+
+
+def remove_spaces(s):
+  s = s.replace(' ', '')
+  s = s.replace('\n', '')
+  return s
```

Comment:
I'd rather use a regex here:

`return re.sub('\s+', ' ', s)`

Reply:
That's right, it's better
You meant `return re.sub('\s+', '', s)` to remove the spaces right ?

Reply:
Yeah, should be an empty replacement string.